### PR TITLE
Issue 209- Better Documentation for .js

### PR DESCRIPTION
**Pull request by KamBustos**
There are undocumented functions/code blocks that need documentation. Future teams can use this documentation to either help work on future issues, or better understand the codebase. When it comes to specific files that need this documentation, I would say the majority of them contain some form of functions/code blocks that can use new/extra details.

Files Currently Made Documentation to:

auth.js
deer-config.js
deer-record.js
deer-render.js
glossModal.js
jwt.js
layout.js
mark.min.js
ng.js
plimit.js
text-witness-line-selector.js
tpen-line-selector.js
witnessModal.js

Files that don't need documentation:

deer-utils.js
deer.js
gloss-transcription.js
gloss-witness.js
hello-world.js ( I don't even know what the purpose of this file serves)
manageGlossModal.js
openseadragon.js
openseadragon.js.map
openseadragon.min.js
openseadragon.min.js.map
worker.js (there is nothing in this file?)
yocto.js

### CODE REVIEW

**Review by thehabes — APPROVED** [review]
Big fan, this is tough documentation to do.  I think this is something you teammates should approve as well to be sure the documentation you've added is accurate.  The one who knows that the best is the code author.

File: js/deer-config.js [hunk 1]
```diff
@@ -985,14 +1004,22 @@ function userHasRole(roles){
...
 function debounce(func,timeout = 500) {
     let timeRemains
     return (...args) => {
         clearTimeout(timeRemains)
         timeRemains = setTimeout(()=>func.apply(this,args),timeRemains)
     }
 }
-
+/**
+ * Hides the search submit button when the search bar is empty, reducing visual clutter.
```
Comment (xXCrash2BomberXx) [inline js/deer-config.js:1021]: This suggests that the function is responsible for hiding the search bar but in reality, the function is attaching handlers that do it. The documentation suggests the function should be called every time it needs hidden but it should only be called once

**Review by TheTuTran — COMMENTED** [review]
Looks good overall. Some things might be a little extra like in layout.js or witnessModal.js where you describe the variables and css styling. I feel like it probably isn't necessary. I think just a simple explanation of what the custom element is and the usage of it is good enough.

**Review by AbhilashKotha — APPROVED** [review]
lgtm

**Pull request merged**